cluster-autoscaler/utils/drain/drain.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package drain
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/kubelet/types"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// GetPodsForDeletionOnNodeDrain returns pods that should be deleted on node drain as well as some extra information
+// about possibly problematic pods (unreplicated and deamon sets).
+func GetPodsForDeletionOnNodeDrain(
+	podList []*api.Pod,
+	decoder runtime.Decoder,
+	skipNodesWithSystemPods bool,
+	skipNodesWithLocalStorage bool,
+	checkReferences bool,
+	client *client.Client,
+	minReplica int32) ([]*api.Pod, error) {
+
+	pods := make([]*api.Pod, 0)
+
+	for _, pod := range podList {
+		if IsMirrorPod(pod) {
+			continue
+		}
+
+		daemonsetPod := false
+		replicated := false
+
+		sr, err := CreatorRef(pod)
+		if err != nil {
+			return []*api.Pod{}, fmt.Errorf("failed to obtain refkind: %v", err)
+		}
+		refKind := ""
+		if sr != nil {
+			refKind = sr.Reference.Kind
+		}
+
+		if refKind == "ReplicationController" {
+			if checkReferences {
+				rc, err := client.ReplicationControllers(sr.Reference.Namespace).Get(sr.Reference.Name)
+				// Assume a reason for an error is because the RC is either
+				// gone/missing or that the rc has too few replicas configured.
+				// TODO: replace the minReplica check with pod disruption budget.
+				if err == nil && rc != nil {
+					if rc.Spec.Replicas < minReplica {
```

Comment:
As @fgrzadkowski suggested how about checking here that we are not removing all pods from the given ReplicationController rather than whether the pod is just replicated?